src/authStrategies/RemoteAuth.js

```diff
@@ -0,0 +1,221 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const unzipper = require('unzipper');
+const archiver = require('archiver');
+const BaseAuthStrategy = require('./BaseAuthStrategy');
+
+/**
+ * Remote-based authentication
+ * @param {object} options - options
+ * @param {object} options.store - Remote database store instance
+ * @param {string} options.clientId - Client id to distinguish instances if you are using multiple, otherwise keep null if you are using only one instance
+ * @param {string} options.dataPath - Change the default path for saving session files, default is: "./.wwebjs_auth/" 
+ * @param {number} options.backupSyncMs - Sets the time interval for periodic session backups. Accepts values starting from 60000ms {1 minute}
+ */
+class RemoteAuth extends BaseAuthStrategy {
+    constructor({ clientId, dataPath, store, backupSyncMs } = {}) {
+        super();
+
+        const idRegex = /^[-_\w]+$/i;
+        if (clientId && !idRegex.test(clientId)) {
+            throw new Error('Invalid clientId. Only alphanumeric characters, underscores and hyphens are allowed.');
+        }
+        if (!backupSyncMs || backupSyncMs < 60000) {
+            throw new Error('Invalid backupSyncMs. Accepts values starting from 60000ms {1 minute}.');
```

Comment:
The backup speed process varies depending on the size of the session. Heavy sessions could take even up to 3 minutes to execute completely (or even more and consumes resources). If this process is yet to be completed and another sync is triggered, it is very likely to cause conflicts not only on a process execution level but also on the saving of the session. 

I determined 1 minute as minimum interval to avoid users getting weird errors, even though I personally do not recommend using such a low sync frequency. I'm open to suggestions on adapting the minimum value if u consider it pertinent.